src/main/java/it/gov/pagopa/payhub/activities/activity/exportflow/ExportFileExpirationHandlerActivityImpl.java

```diff
@@ -0,0 +1,83 @@
+package it.gov.pagopa.payhub.activities.activity.exportflow;
+
+import it.gov.pagopa.payhub.activities.connector.processexecutions.ExportFileService;
+import it.gov.pagopa.payhub.activities.exception.exportflow.ExportFileNotFoundException;
+import it.gov.pagopa.payhub.activities.util.AESUtils;
+import it.gov.pagopa.payhub.activities.util.FileShareUtils;
+import it.gov.pagopa.pu.processexecutions.dto.generated.ExportFile;
+import it.gov.pagopa.pu.processexecutions.dto.generated.ExportFileStatus;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@Lazy
+public class ExportFileExpirationHandlerActivityImpl implements
+    ExportFileExpirationHandlerActivity {
+
+  /**
+   * the shared folder
+   */
+  private final Path sharedDirectoryPath;
+
+  private final ExportFileService exportFileService;
+
+  public ExportFileExpirationHandlerActivityImpl(
+      @Value("${folders.shared}") String sharedFolder, ExportFileService exportFileService) {
+    this.sharedDirectoryPath = Path.of(sharedFolder);
+    this.exportFileService = exportFileService;
+
+    if (!Files.exists(sharedDirectoryPath)) {
+      throw new IllegalStateException("Shared folder doesn't exist: " + sharedDirectoryPath);
+    }
+  }
+
+  @Override
+  public void handleExpiration(Long exportFileId) {
+    log.info("Handling expiration of Export File having exportFileId {}", exportFileId);
+    ExportFile file = exportFileService.findById(exportFileId)
+        .orElseThrow(() -> new ExportFileNotFoundException(
+            "Export file having exportFileId %s not found.".formatted(exportFileId)));
+
+    Path exportFilePath = getFilePath(file);
+    if (exportFilePath != null) {
```

Comment:
let's log on the else branch as INFO that this field was null, print its ID, its status and its type